### PR TITLE
[core] Introduce InPredicateVisitor to refactor Predicate handle

### PR DESCRIPTION
**Pull request by xuzifu666**
### Purpose

Linked issue: close #xxx
from: https://github.com/apache/paimon/pull/4469
1. Follow up https://github.com/apache/paimon/pull/4436 , refactor related logic to a util class to duplicate codes;
2. Fix incidental ut failed due to order in schema test.

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/table/system/TagsTable.java [hunk 1]
```diff
@@ -239,25 +240,18 @@ public RecordReader<InternalRow> createReader(Split split) {
...
+                        if (leafs.isPresent()) {
+                            leafs.get().stream()
+                                    .forEach(
+                                            leaf -> {
+                                                if (tagManager.tagExists(leaf.toString())) {
+                                                    predicateMap.put(
+                                                            leaf.toString(),
+                                                            tagManager.tag(leaf.toString()));
+                                                }
+                                            });
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/table/system/TagsTable.java:254]: `InPredicateVisitor.extractInElements(predicate, TAG_NAME)
                                .ifPresent(leafs -> leafs
                                .forEach(
                                        leaf -> {
                                            String leftName = leaf.toString();
                                            if (tagManager.tagExists(leftName)) {
                                                predicateMap.put(
                                                        leftName,
                                                        tagManager.tag(leftName));
                                            }
                                        }));`

File: paimon-core/src/main/java/org/apache/paimon/table/system/SnapshotsTable.java [hunk 2]
```diff
@@ -231,18 +231,11 @@ public InnerTableRead withFilter(Predicate predicate) {
...
-                        } else {
-                            snapshotIds.clear();
-                            break;
-                        }
-                    }
+                Optional<List<Object>> leafs =
+                        InPredicateVisitor.extractInElements(predicate, leafName);
+                if (leafs.isPresent()) {
+                    leafs.get().stream()
+                            .forEach(leaf -> snapshotIds.add(Long.parseLong(leaf.toString())));
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/table/system/SnapshotsTable.java:238]: InPredicateVisitor.extractInElements(predicate, leafName)
                        .ifPresent(leafs -> leafs
                        .forEach(leaf -> snapshotIds.add(Long.parseLong(leaf.toString()))));

File: paimon-core/src/main/java/org/apache/paimon/table/system/SchemasTable.java [hunk 3]
```diff
@@ -229,17 +230,11 @@ public InnerTableRead withFilter(Predicate predicate) {
...
-                            schemaIds.add((Long) ((LeafPredicate) leaf).literals().get(0));
-                        } else {
-                            schemaIds.clear();
-                            break;
-                        }
+                    Optional<List<Object>> leafs =
+                            InPredicateVisitor.extractInElements(predicate, leafName);
+                    if (leafs.isPresent()) {
+                        leafs.get().stream()
+                                .forEach(leaf -> schemaIds.add(Long.parseLong(leaf.toString())));
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/table/system/SchemasTable.java:237]: `InPredicateVisitor.extractInElements(predicate, leafName)
                            .ifPresent(leafs -> leafs
                            .forEach(leaf -> schemaIds.add(Long.parseLong(leaf.toString()))));`

File: paimon-common/src/main/java/org/apache/paimon/predicate/InPredicateVisitor.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/** A utils to handle {@link Predicate}. */
+public class InPredicateVisitor {
+
+    /**
+     * Method for handling with CompoundPredicate.
```
Comment (wwj6591812) [inline paimon-common/src/main/java/org/apache/paimon/predicate/InPredicateVisitor.java:29]: Method for handling with In CompoundPredicate

**Comment by xuzifu666** [comment]
@wwj6591812 had addressed, Thanks.

**Comment by wwj6591812** [comment]
+1

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**